### DISCUSSION TITLE
middleware "https-redirectscheme@file" does not exist

### DISCUSSION THREAD

**Discussion by wolfman2g1** [techno-tim/techno-tim.github.io#174] [Questions] [answered]
https://github.com/techno-tim/techno-tim.github.io/blob/9ecda730a9a0a02476a98ef734abd092f1dd219b/reference_files/traefik-portainer-ssl/traefik/config.yml#L10

Pardon, 
I'm more familiar with Nginx, I've been getting error in the logs. I've done some googling and tried

``middlewares:
        - default-headers
        - middlewares-https-redirectscheme`
       `middlewares:
    addprefix-pihole:
      addPrefix:
        prefix: "/admin"
    middlewares-https-redirectscheme:
      redirectScheme:
        scheme: https
        permanent: true`
  Which also doesn't work

full config
`http:
 #region routers 
  routers:
    proxmox:
      entryPoints:
        - "https"
      rule: "Host(`epyon.example.com`)"
      middlewares:
        - default-headers
        - middlewares-https-redirectscheme
      tls: {}
      service: proxmox
    pihole:
      entryPoints:
        - "https"
      rule: "Host(`dns.example.com`)"
      middlewares:
        - default-headers
        - addprefix-pihole
        - https-redirectscheme
      tls: {}
      service: pihole
    homeassistant:
      # For Homeassistant config, check: https://www.home-assistant.io/integrations/http/#reverse-proxies
      # This relies on Homeassistant using http. No certs are needed in the Homeassistant config.
      entryPoints:
        - "https"
      rule: "Host(`ha.example.com`)"
      middlewares:
        - default-headers
        - https-redirectscheme
      tls: {}
      service: homeassistant
    truenas:
      entryPoints:
        - "https"
      rule: "Host(`sandrock.example.com`)"
      middlewares:
        - default-headers
        - https-redirectscheme
      tls: {}
      service: truenas
    plex:
      entryPoints:
        - "https"
      rule: "Host(`plex.example.com`)"
      middlewares:
        - default-headers
        - https-redirectscheme
      tls: {}
      service: plex
   
    opnsense:
      entryPoints:
        - "https"fw.example.com`)"
      middlewares:
        - default-headers
        - https-redirectscheme
      tls: {}
      service: opnsense
    pterodactyl:
      entryPoints:
        - "https"
      rule: "Host(`game-server.example.com`)"
      middlewares:
        - default-headers
        - https-redirectscheme
      tls: {}
      service: pterodactyl

#endregion
#region services
  services:
    proxmox:
      loadBalancer:
        servers:
          - url: "https://10.0.3.195:8006"
        passHostHeader: true
    pihole:
      loadBalancer:
        servers:
          - url: "http://10.0.3.253:80"
          - url: "http://10.0.3.251:80"
        passHostHeader: true
    homeassistant:
      loadBalancer:
        servers:
          - url: "http://10.0.3.250:8123"
        passHostHeader: true
    truenas:
      loadBalancer:
        servers:
          - url: "http://10.0.3.194"
        passHostHeader: true
    plex:
      loadBalancer:
        servers:
          - url: "http:10.0.3.190:32400"
        passHostHeader: true
    opnsense:
      loadBalancer:
        servers:
          - url: "https://10.0.2.1"
        passHostHeader: true
    #pterodactyl:
    #  loadBalancer:
    #    servers:
    #      - url: "http://192.168.0.110:80"
    #    passHostHeader: true
#endregion
  middlewares:
    addprefix-pihole:
      addPrefix:
        prefix: "/admin"
    middlewares-https-redirectscheme:
      redirectScheme:
        scheme: https
        permanent: true

    default-headers:
      headers:
        frameDeny: true
        browserXssFilter: true
        contentTypeNosniff: true
        forceSTSHeader: true
        stsIncludeSubdomains: true
        stsPreload: true
        stsSeconds: 15552000
        customFrameOptionsValue: SAMEORIGIN
        customRequestHeaders:
          X-Forwarded-Proto: https

    default-whitelist:
      ipWhiteList:
        sourceRange:
        - "10.0.0.0/8"
        - "192.168.0.0/16"
        - "172.16.0.0/12"

    secured:
      chain:
        middlewares:
        - default-whitelist
        - default-headers``

**Comment by wolfman2g1** [comment, marked as answer]
so after some sleep i can confirm that

`- middlewares-https-redirectscheme`
works.

**Reply by kbipinkumar** [reply]
hi where should i add this  option `config.yml` ?

**Reply by kbipinkumar** [reply]
nevermind, needed to be added to `middlewares:` section of `config.yml` like this

```yaml
http:


  middlewares:
    https-redirectscheme:
      redirectScheme:
        permanent: true
        scheme: https
        port: "443"

```
